### PR TITLE
threads: implement support for pthread_condattr

### PR DESCRIPTION
**Pull request by loganek**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -264,6 +268,9 @@ ifeq ($(THREAD_MODEL), posix)
 # Specify the tls-model until LLVM 15 is released (which should contain
 # https://reviews.llvm.org/D130053).
 CFLAGS += -mthread-model posix -pthread -ftls-model=local-exec
+
+# Include cloudlib's directory to access the structure definition of clockid_t
+CFLAGS += -I$(LIBC_BOTTOM_HALF_CLOUDLIBC_SRC)
```
Comment (loganek) [inline Makefile:273]: For `pthread_condattr_setclock` (in top half) we need a definition of the `clockid_t` struct, which is defined in bottom half. Definition of `clockid_t` at least for now is not required to be a part of public interface, therefore I don't think we have to make it public at that stage. At the same time, I'm a bit worried that adding the whole cloudlib source dir as include by default to top half will lead to some weird usage if we're not careful enough. Another option would be to have a dedicated folder with headers shared between bottom and top half (but not available in public interface).

I'm open for suggestions.

Comment (sbc100) [reply Makefile:273]: This solution seems ok to me.

File: libc-top-half/musl/src/thread/pthread_condattr_setclock.c [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
+#include <common/clock.h>
+
 int pthread_condattr_setclock(pthread_condattr_t *a, clockid_t clk)
 {
-	if (clk < 0 || clk-2U < 2) return EINVAL;
+	if (clk->id < 0 || clk->id-2U < 2) return EINVAL;
 	a->__attr &= 0x80000000;
-	a->__attr |= clk;
+	a->__attr |= clk->id;
 	return 0;
```
Comment (sbc100) [inline libc-top-half/musl/src/thread/pthread_condattr_setclock.c:10]: For changes to musl sources like this we wrap them in `__wasilibc_unmodified_upstream`

File: libc-top-half/musl/src/thread/pthread_condattr_setclock.c [hunk 3]
```diff
@@ -1,9 +1,11 @@
 #include "pthread_impl.h"
 
+#include <common/clock.h>
+
 int pthread_condattr_setclock(pthread_condattr_t *a, clockid_t clk)
 {
-	if (clk < 0 || clk-2U < 2) return EINVAL;
+	if (clk->id < 0 || clk->id-2U < 2) return EINVAL;
```
Comment (sbc100) [inline libc-top-half/musl/src/thread/pthread_condattr_setclock.c:7]: I wonder if we can drop `cloudlibc/src/common/clock.h` completely and just use __wasi_clockid_t as the definition for clockid_t?.. avoiding the need for this change?

Comment (sbc100) [reply libc-top-half/musl/src/thread/pthread_condattr_setclock.c:7]: Actually I guess the comments in `libc-bottom-half/cloudlibc/src/common/clock.h` are pretty clear this was a deliberate choice.   I guess this change is needed.

**Review by sbc100 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (loganek) [reply Makefile:273]: ok, let's keep it as it is for now then. Thanks for the feedback.

File: libc-top-half/musl/src/thread/pthread_condattr_setclock.c (hunk 2 above)
Comment (loganek) [reply libc-top-half/musl/src/thread/pthread_condattr_setclock.c:20]: updated

File: libc-top-half/musl/src/thread/pthread_condattr_setclock.c (hunk 3 above)
Comment (loganek) [reply libc-top-half/musl/src/thread/pthread_condattr_setclock.c:12]: ok

**Review by abrown — APPROVED** [review]

**Pull request merged**